Game2048Intelligence/src/main/java/heuristic/GameHeuristic.java

```diff
@@ -0,0 +1,52 @@
+package heuristic;
+
+import models.game2048.Scoreboard;
+import models.grid2048.Grid2048;
+import models.grid2048.Grid2048Event;
+import models.game2048.KeyEventHandler;
+
+public abstract class GameHeuristic {
+    private final Grid2048 grid2048;
+    private final Scoreboard scoreboard;
+    private final KeyEventHandler keyEventUp;
+    private final KeyEventHandler keyEventDown;
+    private final KeyEventHandler keyEventLeft;
+    private final KeyEventHandler keyEventRight;
+
+    public GameHeuristic(Grid2048 grid2048, Scoreboard scoreboard) {
+        this.grid2048 = grid2048;
+        this.scoreboard = scoreboard;
+        this.keyEventUp = Grid2048Event.UP.buildKeyEventHandlerOnKeyEventNoRandomTile(grid2048, scoreboard);
```

Comment:
I've never really seen this before, but I don't think this is a good way to use the enum. See the above related enum changes